apps/wallet/src/ui/app/shared/forms/NumberField.tsx

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type ComponentProps, type ReactNode, forwardRef } from 'react';
+import FormField from './FormField';
+import { Input } from './controls/Input';
+
+type NumberField = {
```

Comment:
instead of having this custom component, I think you should just be able to do `<TextField type="number" <insert other input props here> />`. I guess `TextField` might not have been the best name for that component hahaha